### PR TITLE
refactor: added abstract prompt

### PR DESCRIPTION
**Pull request by PierreDemailly**
This PR add `AbstractPrompt`, `TextPrompt`, `SelectPrompt`& `ConfirmPrompt`
Added `TestingPrompt` & improved/added tests.
Improved TS & doc.

### CODE REVIEW

File: test/text-prompt.test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// Import Internal Dependencies
+import { TextPrompt } from '../src/text-prompt.js'
+import { TestingPrompt } from './helpers/testing-prompt.js'
+
+describe('TextPrompt', () => {
+  it('message should be required', async () => {
+    assert.throws(() => new TextPrompt(12), { name: 'TypeError', message: 'message must be string, number given.' })
+  })
+
+  it('value should be OK', async () => {
```
Comment (tony-go) [inline test/text-prompt.test.js:14]: Could we change the message to something more descriptive?

File: test/text-prompt.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    const textPrompt = await TestingPrompt.TextPrompt('What\'s your name?', 'Joe', (log) => logs.push(log))
+    const input = await textPrompt.question()
+    assert.deepEqual(input, 'Joe')
+    assert.deepEqual(logs, [
+      "? What's your name?",
+      "✔ What's your name? › Joe"
+    ])
+  })
+
+  it('value should be KO', async () => {
```
Comment (tony-go) [inline test/text-prompt.test.js:25]: Could we change the message to something more descriptive?

File: test/select-prompt.test.js [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+      '   Option 2 ',
+      ' › Option 3 ',
+      '   Option 4 ',
+      '⭣  Option 5 ',
+      '⭡  Option 2 ',
+      '   Option 3 ',
+      ' › Option 4 ',
+      '   Option 5 ',
+      '⭣  Option 6 ',
+      '✔ Choose option › Option 4'
```
Comment (tony-go) [inline test/select-prompt.test.js:276]: I confess that it's a bit hard to understand this string; I think that it's due to the scrolling rule.

Could you add comment to explain it?

**Review by tony-go — APPROVED** [review]
👏🏼 Thanks ^^

It's a bit bold, so hard to review :) 

LGTM % comments

> Advice for next time: try to split it in shorter PR; easier to review ^^

**Comment by PierreDemailly** [comment]
Thanks for your time @tony-go 
Yeah I agree the PR is not short and can be hard to review. Not easy to make shorter tho ^^
Tests should be clearer now ;)

**Review by fraxken — APPROVED** [review]

**Pull request merged**